### ISSUE TITLE
`UnnecessaryQualifier` outputs warnings for qualifier annotations copied by lombok

### ISSUE THREAD

**Issue by facboy** [google/error-prone#5128]
Not sure if this is feasible to address, but e.g. with lombok you can do

```
@RequiredArgsConstructor
class MyClass {

   @SpecialString final String injectedString;
}
```

and configure it to copy `@SpecialString` to the generated constructor's parameter.  at the moment errorprone is warning about the qualifier on the injected field, but there's nowhere else to put it in lombok land.

**Comment by bannmann** [comment]
Another example:

``` java
@RequiredArgsConstructor(onConstructor_ = @Inject, access = AccessLevel.PROTECTED)
public final class FooDeleter
{
    @Deleted
    private final Event<Foo> fooDeletedEvent;

    public void delete(int connectionId)
    {
        // ...
    }
}
```
